pyaedt/modeler/cad/Modeler.py

```diff
@@ -1840,6 +1886,140 @@ def set_working_coordinate_system(self, name):
+        to_global : bool, optional
+            Whether to compute the inverse transformation of the input coordinate system with
+            respect to the global coordinate system. The default is ``False``.
+
+        Returns
+        -------
+        list
+            Origin coordinates.
+        list
+            Quaternion.
+        """
+        cs_names = [i.name for i in self.coordinate_systems]
+        if isinstance(coordinate_system, BaseCoordinateSystem):
+            cs = coordinate_system
+        elif isinstance(coordinate_system, str):
+            if coordinate_system not in cs_names:  # pragma: no cover
+                raise AttributeError("Specified coordinate system does not exist in the design.")
+            idx = cs_names.index(coordinate_system)
+            cs = self.coordinate_systems[idx]
+        else:  # pragma: no cover
+            raise AttributeError("coordinate_system must be in either a string or a CoordinateSystem object.")
+
+        if to_global:
+            o, q = self.reference_cs_to_global(coordinate_system)
+            o = [-i for i in GeometryOperators.q_rotation(o, q)]
+            q = [q[0]] + [-i for i in q[1:]]
+        else:
+            q = cs.quaternion
+            q = [q[0]] + [-i for i in q[1:]]
+            o = [-i for i in GeometryOperators.q_rotation(cs.origin, q)]
+        return o, q
+
+    @pyaedt_function_handler()
+    def reference_cs_to_global(self, coordinate_system):
+        """Get the origin and quaternion defining the coordinate system in the global coordinates.
+
+        Parameters
+        ----------
+        coordinate_system : str, CoordinateSystem
+            Name of the destination reference system. The CoordinateSystem object can also be used.
+
+        Returns
+        -------
+        list
+            Origin coordinates.
+        list
+            Quaternion.
+        """
+        cs_names = [i.name for i in self.coordinate_systems]
+        if isinstance(coordinate_system, BaseCoordinateSystem):
+            cs = coordinate_system
+        elif isinstance(coordinate_system, str):
+            if coordinate_system not in cs_names:  # pragma: no cover
+                raise AttributeError("Specified coordinate system does not exist in the design.")
+            idx = cs_names.index(coordinate_system)
+            cs = self.coordinate_systems[idx]
+        else:  # pragma: no cover
+            raise AttributeError("coordinate_system must be in either a string or a CoordinateSystem object.")
+        q = cs.quaternion
+        o = cs.origin
+        ref_cs_name = cs.ref_cs
+        while ref_cs_name != "Global":
+            ref_cs = self.coordinate_systems[cs_names.index(ref_cs_name)]
+            q_ref = ref_cs.quaternion
+            q = GeometryOperators.q_prod(q_ref, q)
+            o_ref = ref_cs.origin
+            o = GeometryOperators.v_sum(o_ref, GeometryOperators.q_rotation(o, q_ref))
+            ref_cs_name = ref_cs.ref_cs
+        return o, q
```

Comment:
If possible, could you rename the variable `o` and `q` to avoid single letter variables.
It would ease the maintenance latter.
```suggestion
        quaternion = cs.quaternion
        origin = cs.origin
        ref_cs_name = cs.ref_cs
        while ref_cs_name != "Global":
            ref_cs = self.coordinate_systems[cs_names.index(ref_cs_name)]
            quaternion_ref = ref_cs.quaternion
            quaternion = GeometryOperators.q_prod(quaternion_ref, quaternion)
            origin_ref = ref_cs.origin
            origin = GeometryOperators.v_sum(origin_ref, GeometryOperators.q_rotation(origin, quaternion_ref))
            ref_cs_name = ref_cs.ref_cs
        return origin, quaternion
```